src/Uno.UWP/Helpers/AppDataUriEvaluator.cs

```diff
@@ -0,0 +1,101 @@
+﻿using System;
+using System.IO;
+using Windows.ApplicationModel;
+using Windows.Storage;
+
+namespace Uno.Helpers
+{
+	/// <summary>
+	/// Logic is based on <see cref="https://docs.microsoft.com/en-us/windows/uwp/app-resources/uri-schemes#ms-appdata" />.
+	/// </summary>
+	internal static class AppDataUriEvaluator
+	{
+		private const string LocalFolderRoute = "local";
+		private const string RoamingFolderRoute = "roaming";
+		private const string TemporaryFolderRoute = "temp";		
+
+		/// <summary>
+		/// Converts given ms-appdata: URI to filesystem path.
+		/// </summary>
+		/// <param name="appdataUri">ms-appdata: URI.</param>
+		/// <returns>Filesystem path.</returns>
+		public static string ToPath(Uri appdataUri)
+		{
+			if (appdataUri == null)
+			{
+				throw new ArgumentNullException(nameof(appdataUri));
+			}
+
+			if (!appdataUri.IsAbsoluteUri)
+			{
+				throw new ArgumentOutOfRangeException(nameof(appdataUri), "URI must be absolute.");
+			}
+
+			if (!appdataUri.Scheme.Equals("ms-appdata", StringComparison.InvariantCultureIgnoreCase))
+			{
+				throw new ArgumentOutOfRangeException(nameof(appdataUri), "URI must be ms-appdata.");
+			}
+
+			var path = appdataUri.AbsolutePath;
+			if (!Path.IsPathRooted(path))
+			{
+				throw new ArgumentOutOfRangeException(nameof(appdataUri), "URI path must be rooted.");
+			}
+
+			var root = Path.GetPathRoot(path);
+			var relativePath = Path.GetRelativePath(root, path);
+
+			var directorySeparator = Path.DirectorySeparatorChar.ToString();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use the overload that takes a 'CultureInfo' or 'IFormatProvider' parameter.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5304285)